syncthing.te

```diff
@@ -0,0 +1,45 @@
+policy_module(syncthing, 1.0.0)
+
+# define types
+type syncthing_t;
+type syncthing_exec_t;
+userdom_user_application_domain(syncthing_t, syncthing_exec_t)
+
+type syncthing_config_home_t;
+userdom_user_home_content(syncthing_config_home_t)
+
+# file permissions
+fs_getattr_xattr_fs(syncthing_t)
+
+userdom_manage_user_home_content(syncthing_t)
+
+# read randomness
+dev_read_rand(syncthing_t)
+dev_read_urand(syncthing_t)
+
+# networking
+userdom_basic_networking(syncthing_t)
+
+kernel_read_system_state(syncthing_t)
+kernel_read_net_sysctls(syncthing_t)
+
+sysnet_dns_name_resolve(syncthing_t)
+
+corenet_tcp_bind_generic_node(syncthing_t)
+corenet_udp_bind_generic_node(syncthing_t)
+
+corenet_tcp_bind_syncthing_port(syncthing_t)
+corenet_tcp_sendrecv_syncthing_port(syncthing_t)
+
+corenet_tcp_bind_syncthing_admin_port(syncthing_t)
+corenet_tcp_sendrecv_syncthing_admin_port(syncthing_t)
+
+corenet_udp_bind_syncthing_discovery_port(syncthing_t)
+corenet_udp_sendrecv_syncthing_discovery_port(syncthing_t)
+
+# file transitions
+syncthing_config_filetrans(syncthing_t)
```

Comment:
On 08/21/2016 11:09 PM, Naftuli Tzvi Kay wrote:
>> +sysnet_dns_name_resolve(syncthing_t)
>> +
>> +corenet_tcp_bind_generic_node(syncthing_t)
>> +corenet_udp_bind_generic_node(syncthing_t)
>> +
>> +corenet_tcp_bind_syncthing_port(syncthing_t)
>> +corenet_tcp_sendrecv_syncthing_port(syncthing_t)
>> +
>> +corenet_tcp_bind_syncthing_admin_port(syncthing_t)
>> +corenet_tcp_sendrecv_syncthing_admin_port(syncthing_t)
>> +
>> +corenet_udp_bind_syncthing_discovery_port(syncthing_t)
>> +corenet_udp_sendrecv_syncthing_discovery_port(syncthing_t)
>> +
>> +# file transitions
>> +syncthing_config_filetrans(syncthing_t)
> 
> Syncthing is allowed to manage the content via `userdom_manage_user_home_content(syncthing_t)` above.
> 

No that interface only applies to generic user home content
(user_home_t), and thus not private home content associated with type
syncthing_config_home_t

Try this:

allow syncthing_t syncthing_config_home_t:manage_dir_perms;
allow syncthing_t syncthing_config_home_t:manage_file_perms;

-- 
Key fingerprint = 5F4D 3CDB D3F8 3652 FBD8  02D5 3B6C 5F1D 2C7B 6B02
https://sks-keyservers.net/pks/lookup?op=get&search=0x3B6C5F1D2C7B6B02
Dominick Grift